dask/dataframe/groupby.py

```diff
@@ -1212,11 +1212,18 @@ def _groupby_apply_funcs(df, *by, **kwargs):
 def _compute_sum_of_squares(grouped, column):
     # Note: CuDF cannot use `groupby.apply`.
     # Need to unpack groupby to compute sum of squares
-    if hasattr(grouped, "grouper"):
-        keys = grouped.grouper
-    else:
-        # Handle CuDF groupby object (different from pandas)
-        keys = grouped.grouping.keys
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            "ignore",
+            "DataFrameGroupBy.grouper is deprecated and will be removed in a future version of pandas.",
+            FutureWarning,
+        )
+        # TODO: Avoid usage of grouper
```

Comment:
How urgent is this? If we ought to get it done soonish maybe it's worth an issue for the backlog?

Reply:
pandas 3.0, not super urgent

It will only warn for now, this whole function is weird

I'll open an issue

Reply:
it looks like a bandaid for cudf, but not sure